deepcell_toolbox/utils_test.py

```diff
@@ -272,3 +273,100 @@ def test_resize():
+
+    for shape, input_shape, stride_ratio, dtype in prod:
+        big_image = (np.random.random(shape) * 100).astype(dtype)
+        tiles, tiles_info = utils.tile_image(
+            big_image, input_shape,
+            stride_ratio=stride_ratio)
+
+        assert tiles.shape[1:] == input_shape + (shape[-1],)
+        assert tiles.dtype == big_image.dtype
+
+        ceil = lambda x: int(np.ceil(x))
+        round_to_even = lambda x: int(np.ceil(x / 2.0) * 2)
+
+        image_size_x, image_size_y = big_image.shape[1:3]
+        tile_size_x = input_shape[0]
+        tile_size_y = input_shape[1]
+
+        stride_x = round_to_even(stride_ratio * tile_size_x)
+        stride_y = round_to_even(stride_ratio * tile_size_y)
+
+        if stride_x > tile_size_x:
+            stride_x = tile_size_x
+        if stride_y > tile_size_y:
+            stride_y = tile_size_y
+
+        rep_number_x = ceil((image_size_x - tile_size_x) / stride_x + 1)
+        rep_number_y = ceil((image_size_y - tile_size_y) / stride_y + 1)
+
+        expected_batches = big_image.shape[0] * rep_number_x * rep_number_y
+
+        assert tiles.shape[0] == expected_batches
+
+    # test bad input shape
+    bad_shape = (21, 21, 1)
+    bad_image = (np.random.random(bad_shape) * 100)
+    with pytest.raises(ValueError):
+        utils.tile_image(bad_image, (5, 5), stride_ratio=0.75)
+
+
+def test_untile_image():
+    shapes = [
+        (1, 256, 256, 1),
+        (4, 300, 300, 2),
+    ]
+    rand_rel_diff_thresh = 1e-2
+    model_input_shapes = [(51, 51), (64, 64), (256, 256), (300, 300)]
+    stride_ratios = [0.33, 0.5, 0.66, 0.75, 1]
+    dtypes = ['int32', 'float32', 'uint16', 'float16']
+    prod = product(shapes, model_input_shapes, stride_ratios, dtypes)
+
+    # Test that randomly generated arrays are unchanged within a moderate tolerance
+    for shape, input_shape, stride_ratio, dtype in prod:
+
+        big_image = (np.random.random(shape) * 100).astype(dtype)
+        tiles, tiles_info = utils.tile_image(big_image, model_input_shape=input_shape,
+                                             stride_ratio=stride_ratio)
+
+        untiled_image = utils.untile_image(tiles, tiles_info)
+
+        assert untiled_image.dtype == dtype
+        assert untiled_image.shape == shape
+
+        np.testing.assert_allclose(big_image, untiled_image, rand_rel_diff_thresh)
+
+    # Test that arrays of zeros are unchanged by tile/untile
+    for shape, input_shape, stride_ratio, dtype in prod:
+
+        big_image_zeros = np.zeros(shape).astype(dtype)
+        tiles, tiles_info = utils.tile_image(big_image_zeros, model_input_shape=input_shape,
+                                             stride_ratio=stride_ratio)
+        untiled_image_zeros = utils.untile_image(tiles, tiles_info)
+        assert untiled_image_zeros.dtype == dtype
+        assert untiled_image_zeros.shape == shape
+        assert big_image_zeros == untiled_image_zeros
```

Comment:
```suggestion
        assert np.testing.assert_equal(big_image_zeros, untiled_image_zeros)
```

Let's use the `np.testing` function instead of the `==` operator.